pvanalytics/quality/irradiance.py

```diff
@@ -660,3 +660,271 @@ def calculate_component_sum_series(solar_zenith,
     return _fill_nighttime(component, component_sum_df,
                            fill_night_value,
                            solar_zenith, zenith_limit)
+
+
+def _upper_poa_global_limit_lorenz(aoi, solar_zenith, dni_extra):
+    r"""Function to calculate the upper limit of poa_global
+    """
+    # Changing aoi to 90 degrees when solar zenith is greater than 90 (sun
+    # below horizon) or aoi is greater than 90 (sun on the other side of
+    # the sensor/module's plane).
+    aoi = aoi.clip(lower=0, upper=90)
+
+    # Determining the upper limit
+    upper_limit = 0.9 * dni_extra * (cosd(aoi))**1.2 + 300
+
+    # Setting upper limit as 0 when solar zenith is > 90 (night time)
+    upper_limit[solar_zenith > 90] = 0
+
+    # Setting upper limit as undefined where solar_zenith is not available
+    upper_limit[solar_zenith.isna()] = np.nan
+
+    # Setting upper limit as undefined where aoi is not available
+    upper_limit[aoi.isna()] = np.nan
+
+    return upper_limit
+
+
+def _lower_limit_lorenz(solar_zenith, dni_extra):
+    r"""Function to calculate the lower limit of poa_global and ghi
+    """
+    # Setting the lower_limit at 0.
+    lower_limit = pd.Series(np.zeros(len(solar_zenith)),
+                            index=solar_zenith.index)
+
+    # Determining the lower limit when solar zenith is < 75
+    lower_limit = lower_limit.mask(solar_zenith < 75,
+                                   0.01 * dni_extra * cosd(solar_zenith))
+
+    # Setting lower limit as undefined where solar_zenith is not available
+    lower_limit[solar_zenith.isna()] = np.nan
+
+    return (lower_limit)
+
+
+def check_poa_global_limits_lorenz(poa_global, solar_zenith, aoi,
+                                   dni_extra=1367):
+    r"""Test for limits on POA global using the equations described in
+    Section 6.1 of [1]_
+
+    Criteria from [1]_ are used to determine physically plausible
+    lower and upper bounds. Each value is tested and a value passes if
+    value > lower bound and value < upper bound. Also, steps with
+    change in magnitude of more than 1000 W/m2 are flagged. Lower bounds are
+    constant for all tests. Upper bounds are calculated as
+
+    .. math::
+        upper\_limit = 0.9 * dni\_extra * cos(aoi)^{1.2} + 300
+
+    Parameters
+    ----------
+    poa_global : Series
+        Global tilted irradiance in :math:`W/m^2`
+    solar_zenith : Series
+        Solar zenith angle in degrees
+    aoi : Series
+        Direct normal irradiance in :math:`W/m^2`
+    dni_extra : float
+        normal irradiance at the top of atmosphere in :math:`W/m^2`
```

Comment:
```suggestion
    poa_global : Series
        Global tilted irradiance [W/m^2]
    solar_zenith : Series
        Solar zenith angle [degrees]
    aoi : Series
        angle of incidence [degrees]
    dni_extra : float, default 1367
        normal irradiance at the top of atmosphere [W/m^2]
```

Reply:
I copied the math type from `check_ghi_limits_qcrad` function. May be a silly question, but to avoid such future changes, what style can I refer to when documenting such functions?

Reply:
Yes, sorry that pvanalytics is inconsistent.  This is the same style we're using in pvlib (although older code in pvlib is also inconsistent).  Someday we will go through and standardize both packages on our current style.  It's mostly  [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) but with some of our own conventions like this way of formatting units.  I don't think we've documented this style anywhere, but the docstring of [pvlib.temperature.prilliman](https://github.com/pvlib/pvlib-python/blob/v0.9.3/pvlib/temperature.py#L831) is a good example.